File: Semestrul 3/MAP/lab 2/src/Factory/TaskContainerFactory.java
package Factory;

import Containers.Container;
import Containers.QueueContainer;
import Containers.StackContainer;

public class TaskContainerFactory implements Factory{
    @Override
    public Container createContainer(Strategy strategy) {
        switch (strategy){
            case FIFO -> { return new QueueContainer(); }
            case LIFO -> { return new StackContainer(); }
        }
        return null;
    }
}

File: Semestrul 3/MAP/lab 10/factory/TaskContainerFactory.cs
namespace lab10.factory;

public class TaskContainerFactory : IFactory
{
    public IContainer? CreateContainer(Strategy strategy)
    {
        return strategy switch
        {
            Strategy.Fifo => new QueueContainer(),
            Strategy.Lifo => new StackContainer(),
            _ => null
        };
    }
}